clang/lib/APINotes/APINotesYAMLCompiler.cpp

```diff
@@ -730,7 +731,9 @@ class YAMLConverter {
     }
   }
 
-  void convertParams(const ParamsSeq &Params, FunctionInfo &OutInfo) {
+  std::optional<ParamInfo> convertParams(const ParamsSeq &Params,
+                                         FunctionInfo &OutInfo) {
```

Comment:
While currently we only apply a note to a single implicit parameter (`this` or `self`), what if we were to extend this in the future? Does it make sense to take a reference to a `std::optional<ParamInfo>` that we setup as the implicit instance parameter?

Reply:
I think we can do that change when we need additional implicit parameters (or make it return a struct instead of a `std::optional<ParamInfo>`. One of the main reasons why I prefer returning an optional at the moment because it makes some of the call sites more concise, e.g., the checking for trying to specify `self` or `this` for a free function.

Reply:
I suppose that is fine. I was suggesting that we keep the return type `void` and instead take an out parameter that is filled in. That would still keep the call site concise (IMO) and still make it easier to extend in the future (at least because there will be a design in place already which can be extended).

Reply:
Updated the PR.